topic.go

```diff
@@ -0,0 +1,115 @@
+package kafkalock
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+func validateTopic(config *Config) error {
+	if config.TopicPolicy == TopicPolicyNoValidation {
+		return nil
+	}
+
+	kafkaConfig := *config.KafkaConfig
+	kafkaConfig.Metadata.Full = true
+	admin, err := sarama.NewClusterAdmin(config.BootstrapServers, &kafkaConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create cluster admin: %w", err)
+	}
+	defer admin.Close()
+
+	topics, err := admin.ListTopics()
+	if err != nil {
+		return fmt.Errorf("failed to list topics: %w", err)
+	}
+
+	topicDetail := &sarama.TopicDetail{
+		NumPartitions:     int32(config.MaxHolders),
+		ReplicationFactor: 1,
```

Comment:
Don't we want 3 / + make it configurable?  What if one broker goes down?

Reply:
Yeah, I double-checked the docs ... I misunderstood it at first. I will make it configurable.